packages/records/lib/models/records.ts

```diff
@@ -199,6 +200,7 @@ export async function getRecords({
             const decryptedData = await decryptRecordData(item);
             results.push({
                 ...decryptedData,
+                id: item.external_id, // record payload can be empty (when pruned), always use external_id as id
```

Comment:
doesn't change anything for non-pruned records (`payload.id == external_id`) but necessary for pruned records where the payload is empty